praetorian_cli/ui/aegis/menu.py

```
@@ -139,32 +182,224 @@ def _get_schedule_completions(self, prefix):
+
+def _fetch_remote_files(menu, cache_key, user, public_hostname, directory):
+    """Background worker: SSH into the agent and cache the directory listing."""
+    lock = getattr(menu, '_remote_ls_lock', None)
+    try:
+        ssh_cmd = [
+            'ssh', '-o', 'ConnectTimeout=3', '-o', 'StrictHostKeyChecking=accept-new',
+            '-o', 'BatchMode=yes',
+            f'{user}@{public_hostname}',
+            f'ls -1F {directory}',
+        ]
+        result = subprocess.run(
```

Ruff (0.15.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)